Automattic/newspack-plugin
# feat(ia): audience configuration boilerplate

### All Submissions:


### Changes proposed in this Pull Request:

Adds **Audience > Configuration** Wizard. Existing React components were migrated to new directory (for the most part) as-is, the only exception is in the way Audience related wizards are presented using dynamic imports. 

Adds **Audience > Campaign** boilerplate. This was done for 3 reasons:
1. To provide an example of implementing further Audience related wizards.
2. Supply a destination for call to actions within Audience > Configuration.
3. If a admin menu (i.e. Audience) only has one submenu, no submenus are shown.

#### New menu & submenus:
![Screenshot 2024-11-06 at 12 26 13](https://github.com/user-attachments/assets/91192bf4-c1d6-43b5-ba22-1b5ac5d409dc)

#### RAS:
| Inactive     | Active      |
| ------------- | ------------- |
| ![Screenshot 2024-11-07 at 09 23 53](https://github.com/user-attachments/assets/e4f6556d-0ed8-480c-a4fd-9a3e882446f3) | ![Screenshot 2024-11-07 at 09 35 33](https://github.com/user-attachments/assets/891a2f82-a590-45fd-9367-50c853095d90) |

#### Important Information:
When comparing to Figma, there are a number of difference, mainly around missing settings namely; Platform, Salesforce and Stripe components - these will be added in a subsequent PR. When testing "Reader Revenue" prerequisite its important to remember that this UX will change, in that, instead of clicking through to a page it will open a Modal with all related settings i.e. Platform etc.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets `git checkout origin/feat/ia-audience && npm run build`
2. It's best to test with as little of the Reader activation prerequisites active as possible, run the following CLI command to reset most Reader Activation prerequisites (depending on what you have configured locally you should see something resembling RAS > "Inactive" (above)): 
```sh
wp option delete newspack_reader_activation_active_campaign_master_list newspack_reader_activation_contact_email_address newspack_reader_activation_enabled newspack_reader_activation_mailchimp_audience_id newspack_reader_activation_mailchimp_reader_default_status newspack_reader_activation_newsletter_lists newspack_reader_activation_newsletters_label newspack_reader_activation_sender_email_address newspack_reader_activation_sender_name newspack_reader_activation_sync_esp newspack_reader_activation_terms_text newspack_reader_activation_terms_url newspack_reader_activation_use_custom_lists _newspack_ras_skip_campaign_setup
```
3. Step through and active each prerequisite. Ensure call-to-actions and back links navigate to correct locations and data persists.
4. Once all prerequisites are active go ahead and test settings underneath "Show Advanced Settings".

### Other information:
